File: 2559-count-vowel-strings-in-ranges/2559-count-vowel-strings-in-ranges.ts
function isVowelWord(word: string): boolean {
    let vowels = ['a', 'e', 'i', 'o', 'u']

    return vowels.includes(word[0]) && vowels.includes(word[word.length - 1])
}

function vowelStrings(words: string[], queries: number[][]): number[] {
    let n = words.length

    // Create the pefix sum array
    let prefix_sum = new Array(n + 1).fill(0)

    // Adding the prefix sum to the prefix sum array
    for (let i = 0; i < n; i++) {
        prefix_sum[i + 1] = prefix_sum[i] + (isVowelWord(words[i]) ? 1 : 0)
    }

    let results: number[] = []

    for(const [l,r] of queries){
        results.push(prefix_sum[r + 1] - prefix_sum[l])
    }

    return results

};

File: 2559-count-vowel-strings-in-ranges/2559-count-vowel-strings-in-ranges.py
class Solution:
    def isVowelString(self, word: str) -> bool:
        vowels = { 'a', 'e', 'i', 'o', 'u' }
        return word[0] in vowels and word[-1] in vowels
        
    def vowelStrings(self, words: List[str], queries: List[List[int]]) -> List[int]:
        n = len(words)
        
        # Create prefix sum array
        prefix_sum: List[int] = [0] * (n + 1)
        
        # Adding the prefix sum to the array
        for i in range(n):
            prefix_sum[i + 1] = prefix_sum[i] + (1 if self.isVowelString(words[i]) else 0)
        
        # Prcoess each query
        results: List[int] = []
        
        for l, r in queries:
            results.append(prefix_sum[r + 1] - prefix_sum[l])
            
        return results
        
        